source/store/operations_and_invariants/other_invariants.py

```diff
@@ -157,6 +157,29 @@ def print(graph, precision):
         return Utils.print_matrix(SignlessLaplacianDistanceMatrix.calculate(graph), precision)
 
 
+class EccentricityMatrix(InvariantOther):
+    name = "Eccentricity Matrix"
+    type = 'matrix'
+
+    @staticmethod
+    def calculate(graph):
+        if nx.is_connected(graph):
+            size = len(graph.nodes)
+            eccentricity_matrix = np.zeros((size, size))
+            for i in range(size):
+                for j in range(size):
+                    min_eccentricity = min(nx.eccentricity(graph, i), nx.eccentricity(graph, j))
+                    if min_eccentricity == nx.shortest_path_length(graph, i, j):
```

Comment:
Instead of calculating the distance of two vertices several times. First calculate the distance matrix, using the code below. So just look directly at the matrix, which optimizes the code.

`nx.algorithms.shortest_paths.floyd_warshall_numpy(graph)`

Calculating the matrix can be the first step, as it already informs whether the graph is connected.